packages/eslint-plugin-wpcalypso/lib/rules/no-package-relative-imports.js

```diff
@@ -0,0 +1,111 @@
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+const getRelativeImports = ( relativeDir ) => {
+	return fs.readdirSync( relativeDir, { withFileTypes: true } ).map( ( file ) => {
+		const fileName = file.name;
+		if ( file.isDirectory() ) return fileName;
+		const extension = path.extname( fileName );
+		if ( extension !== '.js' ) return fileName;
```

Comment:
We probably need to account for TypeScript `.ts` files as well, and possibly `.json` files too.

Reply:
Good point.
I'll make it configurable, as each project may have their own set of optional extensions.

Reply:
That's an excellent idea 👍

Reply:
what about `.tsx` and `.jsx`?